pravega-cli/src/main/java/io/pravega/tools/pravegacli/commands/admin/PasswordFileCreatorCommand.java

```diff
@@ -0,0 +1,85 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.tools.pravegacli.commands.admin;
+
+import io.pravega.tools.pravegacli.commands.CommandArgs;
+import io.pravega.tools.pravegacli.commands.Command;
+import io.pravega.controller.server.rpc.auth.StrongPasswordProcessor;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.List;
+
+/** The command takes filename and user:password:acl as argument. It creates a file and writes the encrypted password to the file.
+ *
+ */
+public class PasswordFileCreatorCommand extends Command {
+    public PasswordFileCreatorCommand(CommandArgs args){super(args);}
+    private String toWrite;
+
+    public String getToWrite(){
+        return this.toWrite;
+    }
+
+    @Override
+    public void execute() throws InvalidKeySpecException, NoSuchAlgorithmException, IOException {
+        try {
+            ensureArgCount(2);
+            String targetFileName = getTargetFilename(getCommandArgs().getArgs());
+            String userDetails = getUserDetails(getCommandArgs().getArgs());
+            createPassword(userDetails);
+            writeToFile(targetFileName, toWrite);
+        } catch (Exception e){
+            System.err.println(e.getMessage());
+        }
+    }
+
+    public String getTargetFilename(List<String> userInput) {
+        return userInput.get(0);
+    }
+
+    public String getUserDetails(List<String> userInput){
+        String userDetails = userInput.get(1);
+        if((userDetails.split(":")).length == 3){
+            return userDetails;
+        }
+        else throw new IllegalArgumentException("The user detail entered is not of the format uname:pwd:acl");
+    }
+
+    public void createPassword(String userDetails) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        String[] lists = parseUserDetails(userDetails);
+        toWrite = generatePassword(lists);
+    }
+
+    private String[] parseUserDetails(String userDetails){
+        return userDetails.split(":");
+    }
+
+    private String generatePassword(String[] lists) throws NoSuchAlgorithmException, InvalidKeySpecException{
+        StrongPasswordProcessor passwordEncryptor = StrongPasswordProcessor.builder().build();
+        return lists[0] + ":" + passwordEncryptor.encryptPassword(lists[1]) + ":" + lists[2] + ";";
+    }
+
+    private void writeToFile(String targetFileName, String toWrite) throws IOException {
+        try (FileWriter writer = new FileWriter(targetFileName))
+        {
+            writer.write(toWrite + "\n");
+            writer.flush();
+        }
+    }
+
+    public static CommandDescriptor descriptor() {
+        final String COMPONENT = "admin";
```

Comment:
This is a local variable, so use camel case to define it. By the way, what is it actually doing? why it is hardcoded to `admin`?

Reply:
This has the same role as` bk` for all bookkeeper commands. It is part of the `CommandDescriptor` for this command.

Reply:
In all the other commands, this is a `protected` field in an `abstract` class. You are not following this pattern in this PR, but at least define this as an attribute in the class, not as a local variable in capitals (it is confusing).